pkg/crinit/util/util.go

```diff
@@ -18,33 +18,138 @@ package util
 
 import (
 	"fmt"
+	"io"
+	"net"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
 
+	appsv1beta1 "k8s.io/api/apps/v1beta1"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	client "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/cert/triple"
 
+	"github.com/golang/glog"
 	"github.com/spf13/pflag"
 )
 
 const (
 	// DefaultClusterRegistryNamespace is the default namespace in which
 	// cluster registry components are hosted.
 	DefaultClusterRegistryNamespace = "clusterregistry"
+	APIServerCN                     = "clusterregistry"
+	AdminCN                         = "admin"
+	HostClusterLocalDNSZoneName     = "cluster.local."
+	APIServerNameSuffix             = "apiserver"
+	CredentialSuffix                = "credentials"
+
+	lbAddrRetryInterval = 5 * time.Second
+	podWaitInterval     = 2 * time.Second
+
+	apiserverAdvertiseAddressFlag = "api-server-advertise-address"
+	APIServerServiceTypeFlag      = "api-server-service-type"
+	apiserverPortFlag             = "api-server-port"
+
+	apiServerSecurePortName = "https"
+	// Set the secure port to 8443 to avoid requiring root privileges
+	// to bind to port < 1000.  The apiserver's service will still
+	// expose on port 443.
+	apiServerSecurePort = 8443
+)
+
+var (
+	serverName     string
+	serverCredName string
+
+	ComponentLabel = map[string]string{
+		"app": "clusterregistry",
+	}
+
+	apiserverSvcSelector = map[string]string{
+		"app":    "clusterregistry",
+		"module": "clusterregistry-apiserver",
+	}
+
+	apiserverPodLabels = map[string]string{
+		"app":    "clusterregistry",
+		"module": "clusterregistry-apiserver",
+	}
 )
 
 // SubcommandOptions holds the configuration required by the subcommands of
 // `clusterregistry`.
 type SubcommandOptions struct {
-	Name                     string
-	Host                     string
-	ClusterRegistryNamespace string
-	Kubeconfig               string
+	Name                      string
```

Comment:
I wonder if it makes sense to create a specific options package. This file has become very large to be called `util`.

I could see three packages:
- `options`, for options and related
- `common` (ideally with a better name), for functionality that's shared between `standalone` and `aggregated`
- `util`, for things that are specifically utilities

Reply:
Agreed, but for the sake of increasing the size of this PR I left as is. I can follow up with another PR.

Reply:
SGTM.